doc/auth.md

```diff
@@ -0,0 +1,320 @@
+  when: […]
+
+  # the auth rules to apply to the network traffic routed through the targeted resource
+  rules:
+    # (optional) authentication rules to enforce.
+    # at least one config must evaluate to a valid identity object for the auth request to be successful.
+    # if omitted or empty, anonymous access is assumed.
+    authentication:
+      "my-authn-rule":
+        # the authentication method of this rule
+        # one-of: apiKey, jwt, oauth2Introspection, kubernetesTokenReview, x509, plain, anonymous
+        apiKey: {…}
+
+        # (optional) where credentials are required to be passed in the request for authentication based on this rule
+        # one-of: authorizationHeader, customHeader, queryString, cookie
+        credentials:
+          authorizationHeader:
+            prefix: APIKEY
+
+        # (optional) rule-level route selectors
+        routeSelectors: […]
+
+        # (optional) rule-level additional conditions
+        when: […]
+
+        # (optional) configs for caching the resolved object returned out of evaluating this auth rule
+        cache: {…}
+
+    # (optional) rules for fetching auth metadata from external sources.
+    metadata:
+      "my-external-source":
+        # the method for fetching metadata from the external source
+        # one-of: http: userInfo, uma
+        http: {…}
+
+    # (optional) authorization rules to enforce.
+    # all policies must allow access for the auth request be successful.
+    authorization:
+      "my-authz-rule":
+        # the authorization method of this rule
+        # one-of: patternMatching, opa, kubernetesSubjectAccessReview, spicedb
+        opa: {…}
+
+    # (optional) customizations to the authorization response
+    response:
+      # custom denial status and other HTTP attributes for unauthenticated requests.
+      unauthenticated: {…}
+
+      # custom denial status and other HTTP attributes for unauhtorized requests.
+      unauthorized: {…}
+
+      # custom response items when access is granted.
+      success:
+        # custom response items wrapped as HTTP headers to be injected in the request
+        headers:
+          "my-custom-header":
+            # one-of: plain, json, wristband
+            plain: {…}
+
+        # custom response items wrapped as envoy dynamic metadata
+        dynamicMetadata:
+          # one-of: plain, json, wristband
+          "my-custom-dyn-metadata":
+            json: {…}
+
+    # (optional) rules for post-authorization callback requests to external services.
+    # triggered regardless of the result of the authorization request.
+    callbacks:
+      "my-webhook":
+        http: {…}
+```
+
+Check out the [API reference](reference/authpolicy.md) for a full specification of the AuthPolicy CRD.
+
+## Using the AuthPolicy
+
+### Targeting a HTTPRoute networking resource
```

Comment:
Interesting I think I would have inverted these to follow the hierarchy. IE cover gateway first and HTTPRoute second?

Reply:
I see targeting the HTTPRoute as the simplest case and therefore the one to explain first, but maybe that's because of how later we explain, when targeting a gateway, what happens in case there are also other policies targeting HTTPRoutes under that gateway. If we keep that example as-is, then the reader needs to know what targeting a HTTPRoute means beforehand.

Should we go then?

**OPTION 1 (current)**
- _Targeting a HTTPRoute_ – focus on the direct effect of the policy over that route, no mention about Gateway policies
- _Targeting a Gateway_ – explain how the policy affects all routes under the gateway EXCEPT the ones that have a more specific policy of its own

**OPTION 2**
- _Targeting a Gateway_ – explain how it affects all HTTPRoutes under the gateway, no mention about HTTPRoute policies
- _Targeting a HTTPRoute_ – explain how the policy directly affects that route AND replaces any other policy targeting gateways the route is under

Reply:
So I don't see it as a big deal and not a blocker to merge. But my preference is prob Option 2 as it (perhaps I am biased) fits how I think about gateway API resources